okio/src/commonMain/kotlin/okio/internal/HashUtil.kt

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (C) 2018 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package okio.internal
+
+internal infix fun Long.absMod(divisor: Long): Long {
+  val value = this % divisor
+  return if (value < 0) divisor + value else value
+}
+
+internal fun Long.toBigEndianByteArray(): ByteArray = ByteArray(8) { index ->
+  ((this shr ((7 - index) * 8)) and 0xffL).toByte()
+}
+
+internal fun Long.toLittleEndianByteArray(): ByteArray = ByteArray(8) { index ->
+  ((this shr (index * 8)) and 0xffL).toByte()
+}
+
+/**
+ * Left rotate an unsigned 32 bit integer by [bitCount] bits
+ */
+internal infix fun UInt.leftRotate(bitCount: Int): UInt {
+  return ((this shl bitCount) or (this shr (UInt.SIZE_BITS - bitCount))) and UInt.MAX_VALUE
```

Comment:
what does the last `and` operation do? Shouldn’t be necessary?

Reply:
I was planning on removing `left/rightRotate` altogether in favor of the stdlib `rotateRight` and `rotateLeft`. These require stdlib opt in. Are we allowed to do that in Okio?

Reply:
No. We can only depend on stable API.